torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -354,11 +354,38 @@ void TensorPipeAgent::prepareNames() {
   }
 }
 
+void TensorPipeAgent::checkAndSetStaticGroup(
+    const c10::intrusive_ptr<::c10d::Store>& store) {
+  std::string isStaticGroupKey("rpcIsStaticGroup");
+  bool isStaticGroupKeyExists =
+      store->check(std::vector<std::string>{isStaticGroupKey});
+  if (isStaticGroupKeyExists) {
+    std::vector<uint8_t> isStaticGroupVector = store->get(isStaticGroupKey);
+    bool storeIsStaticGroup = false;
+    std::istringstream(std::string(
+        (char*)isStaticGroupVector.data(), isStaticGroupVector.size())) >>
+        std::boolalpha >> storeIsStaticGroup;
+    TORCH_CHECK(
+        isStaticGroup_ == storeIsStaticGroup,
+        fmt::format(
+            "RPC group behavior is different than expected. isStaticGroup_ is initialized with {} while store is {}",
+            isStaticGroup_,
+            storeIsStaticGroup))
+  } else {
+    std::string isStaticGroupStr;
+    isStaticGroupStr = isStaticGroup_ ? "true" : "false";
+    std::vector<uint8_t> isStaticGroupVec(
+        (uint8_t*)isStaticGroupStr.c_str(),
+        (uint8_t*)isStaticGroupStr.c_str() + isStaticGroupStr.length());
+    store->set(isStaticGroupKey, isStaticGroupVec);
+  }
+}
+
 TensorPipeAgent::TensorPipeAgent(
     const c10::intrusive_ptr<::c10d::Store>& store,
     std::string selfName,
     worker_id_t selfId,
-    int worldSize,
```

Comment:
There is no reasoning other than coding style. Since a magic number like -1 or 0 could be used to designate dynamic membership it would be confusing to users; a null value is a bit more explicit. We could also handle and values <= 0 as dynamic.